tests/integration/run_jsonrpc_commands.py

```diff
@@ -5,30 +5,110 @@
 import shlex
 import subprocess
 import sys
+import getopt
 
-def run_shell_command(command: str, expected_response: str) -> int:
+
+def run_shell_command(command: str, expected_response: str, exit_on_fail) -> int:
     """ Run the specified command as shell. """
     command_and_args = shlex.split(command)
-    process = subprocess.run(command_and_args, stdout=subprocess.PIPE, universal_newlines=True, check=True)
+    process = subprocess.run(command_and_args, stdout=subprocess.PIPE,
+                             universal_newlines=True, check=True)
     if process.returncode != 0:
         sys.exit(process.returncode)
     process.stdout = process.stdout.strip('\n')
     response = json.loads(process.stdout)
 
-    if expected_response["result"] is not None and response != expected_response:
-        print("KO: ", command_and_args, " Unexpected result: ", response["result"])
-        sys.exit(1)
     if "error" in response:
-        print("KO: ", command_and_args, " Error: ", response["error"])
-        sys.exit(1)
+        print("--> KO: ", command_and_args, " Error: ", response["error"], "\n")
+        if exit_on_fail:
+            sys.exit(1)
+    elif expected_response["result"] is not None and response != expected_response:
+        print("--> KO: Unexpected Result", command_and_args)
+        print("Response: ", response)
+        print("ExpRsp: ", expected_response, "\n")
+        if exit_on_fail:
+            sys.exit(1)
```

Comment:
See merge conflicts here, changes have just been made in master on this

Reply:
ok